lms/djangoapps/courseware/views/views.py

```diff
@@ -1481,6 +1482,10 @@ def render_xblock(request, usage_key_string, check_if_enrolled=True):
                     'mark-completed-on-view-after-delay': completion_service.get_complete_on_view_delay_ms()
                 }
 
+        if 'application/json' in request.META.get('HTTP_ACCEPT'):
```

Comment:
Should/could we make this a GET parameter instead? That's more explicit and easier to test from browsers. Something like `?format=data`.

Though actually as I think about this more, I think it might be better to make this fully a new API (the "block data API", which has the same interface as the course blocks API but for a subset of the course, or a single block?), or to stick to your original suggestion of making the course blocks API able to handle subsets of the course.

The reason is that this view currently returns a fully rendered XBlock, which is built using all Scope.settings and Scope.user_state data. But `student_view_data` by definition does not include any user-scope data so is not equivalent. It would have to return both `student_view_data` and `student_view_user_state` to be potentially equivalent. But also, all XBlocks can render their student_view but many XBlocks don't define `student_view_data` at all.

Reply:
Okay, in my head I have been thinking of `student_view_data` as returning the dict version of `student_view` and I thought the usage in the course blocks API where the blocks are not bound to a user was just a temporary thing since for most blocks it would not be very useful in an LMS context.?

Yes, I agree the `/blocks/` API design that we created is going to make the most sense. I thought we were waiting for the `ContextLocator` design to be finalized before proposing/implementing that?

Reply:
>  I thought the usage in the course blocks API where the blocks are not bound to a user was just a temporary thing

Nope, by definition, student_view_data never includes any user-specific information, and that's part of the existing API so won't change.

> Yes, I agree the /blocks/ API design that we created is going to make the most sense. I thought we were waiting for the ContextLocator design to be finalized before proposing/implementing that?

Right :) Trying to post my thoughts on that later today.

Reply:
@bradenmacdonald OAuth2 support for this endpoint was requested by @xitij2000 so I have moved this functionality to a new view available at `/xblock/data/<usage_key>`. I have also undone the changes related to `student_view_data`.

Reply:
@symbolist @bradenmacdonald  Sorry maybe I didn't understand the full context here. I just noticed that the credentials were being passed to that URL so assumed a user was needed in the request. If it needs to be authenticated, it might need to support OAuth2 so that we can fetch it from the LMS without needing to share sessions. If it can be fetched without authentication, then it doesn't need OAuth2 support.

Reply:
@xitij2000 This view returns the student_view for blocks so it requires the user to logged in and the credentials to be passed through to LMS. Otherwise, it returns a 404.